js/models/LocalSettings.js

```diff
@@ -3,8 +3,25 @@ import LocalStorageSync from '../utils/backboneLocalStorage';
 import { Model } from 'backbone';
 import is from 'is_js';
 import { feeLevels } from '../utils/fees';
+import { getTranslationLangByCode } from '../data/languages';
 import app from '../app';
 
+/**
+ * Will convert a transifex style language tag to the standard format JS understands as
+ * best as it can. For example, `ca_ES` will be returned as  `ca-ES` and `ca@valencia` will
+ * be returned as `ca`. It's still probably a good idea to wrap any JS that deals with these
+ * codes in a try/catch (e.g. localeCompare) since even a standardized transifex code may not
+ * map to something JS understands.
+ */
+export function standardizedTranslatedLang(lang) {
```

Comment:
This was a bit of a rabbit hole... but since transifex language codes don't necessarily map to ones JS understands, it's important that when you want a language code for JS (e.g. using localeCompare() or Intl.NumberFormat()), rather than obtain it directly from localSettings (`app.localSettings.get('language')`), you retrieve it via `localSettings.standardizedTranslatedLang()`.

That method will try and convert the transifex style to JS style. Keep in mind, it's not a one to one mapping. There are some transifex langs that don't map to ones JS understands (e.g. `tzm`). That's why it's also important that you be cognizant of that fact when using the language in any JS functions (e.g. localeCompare() will throw an exception if it gets an unrecognized lang code, so I needed to explicitly default it to en-US in that case. Intl.NumberFormat() on the other hand was more forgiving and it looks like it defaulted it on its own).